src/NServiceBus.Core/Pipeline/Incoming/IncomingPipelineMetrics.cs

```diff
@@ -239,6 +243,24 @@ public void RecordSendToErrorQueue(IRecoverabilityContext recoverabilityContext)
         totalSentToErrorQueue.Add(1, meterTags);
     }
 
+    public void RecordEnvelopeUnwrappingError(MessageContext messageContext, IEnvelopeHandler type)
+    {
+        if (!totalEnvelopeUnwrappingErrors.Enabled)
+        {
+            return;
+        }
+
+        var incomingPipelineMetricTags = messageContext.Extensions.Get<IncomingPipelineMetricTags>();
+        TagList meterTags;
+        incomingPipelineMetricTags.ApplyTags(ref meterTags, [
+            MeterTags.QueueName,
+            MeterTags.EndpointDiscriminator,
+            MeterTags.MessageType,
+            MeterTags.MessageHandlerType]);
+        meterTags.Add(new KeyValuePair<string, object?>(MeterTags.EnvelopeUnwrapperType, type.GetType().FullName));
```

Comment:
It would be helpful to also add the exception type using the `MeterTags.ErrorType`